staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/accessors.go

```diff
@@ -0,0 +1,149 @@
+	GetMatchPolicy() *v1beta1.MatchPolicyType
+	// GetNamespaceSelector gets the webhook NamespaceSelector field.
+	GetNamespaceSelector() *metav1.LabelSelector
+	// GetSideEffects gets the webhook SideEffects field.
+	GetSideEffects() *v1beta1.SideEffectClass
+	// GetTimeoutSeconds gets the webhook TimeoutSeconds field.
+	GetTimeoutSeconds() *int32
+	// GetAdmissionReviewVersions gets the webhook AdmissionReviewVersions field.
+	GetAdmissionReviewVersions() []string
+
+	// GetMutatingWebhook if the accessor contains a MutatingWebhook, returns it and true, else returns false.
+	GetMutatingWebhook() (*v1beta1.MutatingWebhook, bool)
+	// GetValidatingWebhook if the accessor contains a ValidatingWebhook, returns it and true, else returns false.
+	GetValidatingWebhook() (*v1beta1.ValidatingWebhook, bool)
+}
+
+// MutatingWebhookAccessor is an accessor for a MutatingWebhook.
+type MutatingWebhookAccessor struct {
+	*v1beta1.MutatingWebhook
+}
+
+// ConvertToMutatingWebhookAccessors wraps the provided mutating webhook in a webhook accessor.
+func ConvertToMutatingWebhookAccessors(webhooks []v1beta1.MutatingWebhook) []WebhookAccessor {
+	accessors := make([]WebhookAccessor, len(webhooks))
+	for i, hook := range webhooks {
+		accessors[i] = MutatingWebhookAccessor{&hook}
+	}
+	return accessors
+}
+
+func (m MutatingWebhookAccessor) GetName() string {
+	return m.Name
+}
+func (m MutatingWebhookAccessor) GetClientConfig() v1beta1.WebhookClientConfig {
+	return m.ClientConfig
+}
+func (m MutatingWebhookAccessor) GetRules() []v1beta1.RuleWithOperations {
+	return m.Rules
+}
+func (m MutatingWebhookAccessor) GetFailurePolicy() *v1beta1.FailurePolicyType {
+	return m.FailurePolicy
+}
+func (m MutatingWebhookAccessor) GetMatchPolicy() *v1beta1.MatchPolicyType {
+	return m.MatchPolicy
+}
+func (m MutatingWebhookAccessor) GetNamespaceSelector() *metav1.LabelSelector {
+	return m.NamespaceSelector
+}
+func (m MutatingWebhookAccessor) GetSideEffects() *v1beta1.SideEffectClass {
+	return m.SideEffects
+}
+func (m MutatingWebhookAccessor) GetTimeoutSeconds() *int32 {
+	return m.TimeoutSeconds
+}
+func (m MutatingWebhookAccessor) GetAdmissionReviewVersions() []string {
+	return m.AdmissionReviewVersions
+}
+
+func (m MutatingWebhookAccessor) GetMutatingWebhook() (*v1beta1.MutatingWebhook, bool) {
+	return m.MutatingWebhook, true
+}
+
+func (m MutatingWebhookAccessor) GetValidatingWebhook() (*v1beta1.ValidatingWebhook, bool) {
+	return nil, false
+}
+
+// ValidatingWebhookAccessor is an accessor for a ValidatingWebhook.
+type ValidatingWebhookAccessor struct {
+	*v1beta1.ValidatingWebhook
+}
+
+// ConvertToValidatingWebhookAccessors wraps the provided validating webhook in a webhook accessor.
+func ConvertToValidatingWebhookAccessors(webhooks []v1beta1.ValidatingWebhook) []WebhookAccessor {
+	accessors := make([]WebhookAccessor, len(webhooks))
+	for i, hook := range webhooks {
+		accessors[i] = ValidatingWebhookAccessor{&hook}
```

Comment:
the `hook` var is reused across iterations, so you can't save off the address of it (the same address gets added multiple times, and is overwritten to contain the last iterated item)

should be:
```diff
-       for i, hook := range webhooks {
-               accessors[i] = ValidatingWebhookAccessor{&hook}
+       for i := range webhooks {
+               accessors[i] = ValidatingWebhookAccessor{&webhooks[i]}
        }
```

same comment applies for the mutating constructor

Reply:
I'm going to do some extra reading later to make sure I 100% fully understand why/how go does this iterator pointer thing. Huge thanks for finding this bug.